docs/APIBaselines.md

```diff
@@ -0,0 +1,50 @@
+# API Baselines
+
+This document contains information regarding API baseline files and how to work with them.
+
+## Add baseline files for new projects
+
+When creating a new implementation (i.e. src) project, it's necessary to manually add API baseline files since API baseline are enabled by default. If the project is a non-shipping or test only project, add `<AddPublicApiAnalyzers>false</AddPublicApiAnalyzers>` to the project to disable these checks. To add baseline files to the project:
+
+1. Copy eng\PublicAPI.empty.txt to the project directory and rename it to PublicAPI.Shipped.txt
+1. Copy eng\PublicAPI.empty.txt to the project directory and rename it to PublicAPI.Unshipped.txt
```

Comment:
Are these modified automatically during the build? Is there a way to automatically populate them? If not how should devs update them? Would be worth calling out in the doc

Reply:
Visual Studio is the only reliable way to update the files at the moment. Editing them manually is strongly discouraged because the syntax is finicky.

@Pilchie is working with the Roslyn team to enable `dotnet format` to handle this from the command line.

In the meantime, instructions for adding new implementation projects should mention

1. `cp .\eng\PublicAPI.empty.txt {new folder}\PublicAPI.Shipped.txt`
2. `cp .\eng\PublicAPI.empty.txt {new folder}\PublicAPI.Unshipped.txt`
3. Update AspNetCore.sln and relevant `*.slnf` file to include the new project
3. `{directory containing relevant *.slnf}\startvs.cmd`
4. F6 # or whatever your favourite build gesture is
5. Click on a RS0016 (or whatever) error
6. Right click in editor on underscored symbol or go straight to the “quick fix” icon to its left  . Control-. also works.
7. Choose “Add Blah to public API” / “Fix all occurrences in … Solution”
8. Click Apply
9. F6 # yes, again to see if the fixer missed anything or if other RS00xx errors show up (not uncommon)
10. Suppress or fix other problems as needed but please suppress (if suppressing) using attributes and not globally or with `#pragma`s because attributes make the justification obvious e.g. for common errors that can't be fixed
    `[SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Required to maintain compatibility")]`
    or
    `[SuppressMessage("ApiDesign", "RS0027:Public API with optional parameter(s) should have the most parameters amongst its public overloads.", Justification = "Required to maintain compatibility")]`

More information available in
-	https://github.com/dotnet/roslyn-analyzers/blob/master/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md
-	https://github.com/dotnet/roslyn-analyzers/blob/master/src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md